### PR TITLE
Storage stress test updates

### PR DESCRIPTION
**Pull request by lmolkova**
- run tests on Java21
- use premium storage
- make mem/cpu limit configurable & reduce limits for fault-injector
- use metrics for test results and enable sampling for traces/logs, enable otel manually instead of AppInsights agent
- unify reporting with other stress tests
- update dependency versions

### CODE REVIEW

File: sdk/storage/azure-storage-blob-stress/templates/stress-test-job.yaml [hunk 1]
```diff
@@ -41,8 +41,10 @@ spec:
           {{ ternary "--sync" "" .Stress.sync }} \
           {{ ternary "--faults" "" .Stress.faults }} \
           --warmup 0 \
-          2>&1 | tee -a "${DEBUG_SHARE}/{{ .Stress.testScenario }}-`date +%s`.log";
-
+          2>&1 | tee -a "${DEBUG_SHARE}/{{ .Stress.testScenario }}-`date +%s`.log"; \
+          yum install -y procps; \
+          sleep 10; \
```
Comment (benbp) [inline sdk/storage/azure-storage-blob-stress/templates/stress-test-job.yaml]: @lmolkova the `sleep 10` was just my placeholder for an actual test command, you don't need it.

Also I would move the `procps` install into the Dockerfile if you can.

Comment (lmolkova) [reply sdk/storage/azure-storage-blob-stress/templates/stress-test-job.yaml]: thanks, fixed both

**Review by ibrahimrabab — APPROVED** [review]

File: sdk/storage/azure-storage-blob-stress/Dockerfile [hunk 2]
```diff
@@ -26,17 +26,14 @@ RUN --mount=type=cache,target=/root/.m2 \
  mvn -f /stress/sdk/storage/azure-storage-stress/pom.xml clean install ${SKIP_CHECKS} && \
  mvn -f /stress/sdk/storage/azure-storage-blob-stress/pom.xml clean install ${SKIP_CHECKS}
 
-FROM mcr.microsoft.com/openjdk/jdk:11-mariner
+FROM mcr.microsoft.com/openjdk/jdk:21-mariner
 
 RUN yum -y update
+RUN yum install -y procps
```
Comment (benbp) [inline sdk/storage/azure-storage-blob-stress/Dockerfile:32]:
```suggestion
# Required for `ps` command to find and stop the fault injector process
RUN yum install -y procps
```

File: sdk/storage/azure-storage-blob-stress/scenarios-matrix.yaml [hunk 3]
```diff
@@ -1,13 +1,13 @@
 matrix:
   scenarios:
-#   this test downloads 1024 bytes, small file, no chunking
+    #   this test downloads 1024 bytes, small file, no chunking
```
Comment (benbp) [inline sdk/storage/azure-storage-blob-stress/scenarios-matrix.yaml:3]: Would you prefer to pull out some common fields here, or to keep each test case spec independent? If the former I can help with that.

File: sdk/storage/azure-storage-blob-stress/templates/stress-test-job.yaml [hunk 4]
```diff
@@ -21,35 +24,32 @@ spec:
...
           mkdir -p "$DEBUG_SHARE";
           . /mnt/outputs/.env;
-          export AZURE_HTTP_LOG_DETAIL_LEVEL="HEADERS";
-          export AZURE_LOG_LEVEL="INFO";
-          export APPLICATIONINSIGHTS_INSTRUMENTATION_LOGGING_LEVEL="DEBUG";
-          export APPLICATIONINSIGHTS_ROLE_NAME=storage-{{ .Stress.BaseName }};
           java \
-          -javaagent:BOOT-INF/classes/applicationinsights-agent-3.4.17.jar \
-          -Xms1g \
+          -Dotel.service.name={{ .Release.Name }}-{{ .Stress.BaseName }} \
```
Comment (benbp) [inline sdk/storage/azure-storage-blob-stress/templates/stress-test-job.yaml:28]: Would it be easier to call an external script with these commands? Or does that make it harder to tell what's actually running from the pod if you need to do that?

**Review by benbp — APPROVED** [review]

File: sdk/storage/azure-storage-blob-stress/scenarios-matrix.yaml (hunk 3 above)
Comment (lmolkova) [reply sdk/storage/azure-storage-blob-stress/scenarios-matrix.yaml:3]: common fields would be cool!

Is there a way to reuse the same approach as we have with other test matrices? Like if  I defined one for job generation, I'd do

{
  "resourceProfiles" :  {
      "small": { "cores": 0.5, "memory": "500Mi" },
      "large": { "cores": 2, "memory": "2Gi" },
    },
  "matrix": [
  {
    "resource": "small"
    "sync": [ true, false ],
    "faults":  [ true, false ],
    "blobSize": 1024
    "scenarios": ["downloadcontent", "downloadstream", ...]
  },
  {
    "resource": "large"
    "sync": [ true, false ],
    "faults":  [ true, false ],
    "blobSize": "52428800"
    "scenarios": ["downloadcontent", "downloadstream", ...]
  }
]}

 Could I define something like this in the yaml and do non-sparse generation?

**Pull request merged**